nva/nva.c

```diff
@@ -135,6 +147,84 @@ struct nva_card *nva_init_apu(struct pci_device *dev) {
 	return card;
 }
 
+static int check_modalias(const char *node_name) {
+	char path[PATH_MAX];
+	char buffer[255] = { 0 };
+	FILE *file;
+
+	/* Check if the GPU is indeed made by NVIDIA by checking the compatibility
```

Comment:
The second one is expected. The first one is indeed valid! I will add a check to limit it.

However, we do we check it is a tegra k1 or later? We cannot use the register 0 for that as we do not know the format of their version reg. The modalias attribute could be used to blacklist old platforms as it contains the name of the gpu at least for the gk20a. It will only work on upstream kernels for sure though as downstream ones may not have the same conventions for the modalias. I guess we will not care about those and fix it as needed!